storm-core/src/jvm/org/apache/storm/ui/resources/StormApiResource.java

```diff
@@ -0,0 +1,741 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.storm.ui.resources;
+
+import java.util.HashMap;
+import java.util.Map;
+
+import javax.servlet.ServletContext;
+import javax.servlet.http.HttpServletRequest;
+import javax.ws.rs.Consumes;
+import javax.ws.rs.DefaultValue;
+import javax.ws.rs.GET;
+import javax.ws.rs.POST;
+import javax.ws.rs.Path;
+import javax.ws.rs.PathParam;
+import javax.ws.rs.Produces;
+import javax.ws.rs.QueryParam;
+import javax.ws.rs.core.Context;
+import javax.ws.rs.core.Response;
+import javax.ws.rs.core.SecurityContext;
+import javax.ws.rs.core.UriInfo;
+
+import org.apache.storm.generated.AuthorizationException;
+import org.apache.storm.generated.NotAliveException;
+import org.apache.storm.thrift.TException;
+import org.apache.storm.ui.UIHelpers;
+import org.apache.storm.utils.NimbusClient;
+import org.apache.storm.utils.Utils;
+import org.json.simple.JSONValue;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+
+/**
+ * Root resource (exposed at "storm" path).
+ */
+@Path("/")
+public class StormApiResource {
+
+    public static final Logger LOG = LoggerFactory.getLogger(StormApiResource.class);
+
+    @Context
+    private UriInfo info;
+
+    @Context
+    private HttpServletRequest servletRequest;
+
+
+    @Context
+    private ServletContext servletContext;
+
+    public static Map<String, Object> config = Utils.readStormConfig();
+    /**
+     * /api/v1/cluster/configuration -> nimbus configuration.
+     */
+
+    @GET
+    @Path("/cluster/configuration")
+    @Produces("application/json")
+    public Response getClusterConfiguration() {
+        try {
+            NimbusClient nimbusClient = NimbusClient.getConfiguredClient(Utils.readStormConfig());
```

Comment:
Please put the NimbusClient in the try so it is closed when we are done with it instead of waiting for GC? and we don't have to re-read the stom config for each request.

```
try (NimbusClient nimbusClient = NimbusClient.getConfiguredClient(config)) {
    return Response.status(Response.Status.OK)...
}
```